pkg/router/template/util/util.go

```diff
@@ -47,6 +48,22 @@ func GenerateRouteRegexp(hostname, path string, wildcard bool) string {
 	return "^" + hostRE + portRE + pathRE + subpathRE + "$"
 }
 
+// GenerateSNIRegexp generates a regular expression to match route hosts against
+// a server name in a TLS client hello message.
+func GenerateSNIRegexp(hostname string, wildcard bool) string {
```

Comment:
Could the common code in `GenerateSNIRegexp` and the first half of `GenerateRouteRegexp` be condensed into one "base" function? I think that would make these changes a little easier to follow along with. Also would bring an added benefit of better unit test coverage :smile:

Reply:
Done.  Thanks for the suggestion!